src/atomate2/vasp/sets/base.py

```diff
@@ -2,119 +2,168 @@
-        potcar_spec: bool = False,
-    ) -> None:
-        """Write VASP input files to a directory.
-
-        Parameters
-        ----------
-        directory
-            Directory to write input files to.
-        make_dir
-            Whether to create the directory if it does not already exist.
-        overwrite
-            Whether to overwrite an input file if it already exists.
-        """
-        directory = Path(directory)
-        if make_dir:
-            os.makedirs(directory, exist_ok=True)
-
-        inputs = {"INCAR": self.incar, "KPOINTS": self.kpoints, "POSCAR": self.poscar}
-        inputs.update(self.optional_files)
-
-        if isinstance(self.potcar, Potcar):
-            inputs["POTCAR"] = self.potcar
-        else:
-            inputs["POTCAR.spec"] = "\n".join(self.potcar)
-
-        for key, val in inputs.items():
-            if val is not None and (overwrite or not (directory / key).exists()):
-                with zopen(directory / key, mode="wt") as file:
-                    if isinstance(val, Poscar):
-                        # write POSCAR with more significant figures
-                        file.write(val.get_str(significant_figures=16))
-                    else:
-                        file.write(str(val))
-            elif not overwrite and (directory / key).exists():
-                raise FileExistsError(f"{directory / key} already exists.")
+    structure: Structure | None = None
+    config_dict: dict = field(default_factory=lambda: _BASE_VASP_SET)
+    files_to_transfer: dict = field(default_factory=dict)
+    user_incar_settings: dict = field(default_factory=dict)
+    user_kpoints_settings: dict = field(default_factory=dict)
+    user_potcar_settings: dict = field(default_factory=dict)
+    constrain_total_magmom: bool = False
+    sort_structure: bool = True
+    user_potcar_functional: UserPotcarFunctional = None
+    force_gamma: bool = True
+    reduce_structure: Literal["niggli", "LLL"] | None = None
+    vdw: str | None = None
+    use_structure_charge: bool = False
+    standardize: bool = False
+    sym_prec: float = SETTINGS.SYMPREC
+    international_monoclinic: bool = True
+    validate_magmom: bool = True
+    inherit_incar: bool | list[str] = SETTINGS.VASP_INHERIT_INCAR
+    auto_ismear: bool = True
+    auto_ispin: bool = False
+    auto_lreal: bool = False
+    auto_metal_kpoints: bool = True
+    auto_kspacing: bool = False
+    bandgap_tol: float = SETTINGS.BANDGAP_TOL
+    bandgap: float | None = None
+    prev_incar: str | dict | None = None
+    prev_kpoints: str | Kpoints | None = None
 
     @staticmethod
-    def from_directory(
-        directory: str | Path, optional_files: dict = None
-    ) -> VaspInputSet:
+    def from_directory(directory: str | Path, optional_files: dict = None) -> VaspInput:
         """Load a set of VASP inputs from a directory.
 
         Note that only the standard INCAR, POSCAR, POTCAR and KPOINTS files are read
```

Comment:
any reason to leave `from_directory` and `_get_nedos` here? could both be moved to pymatgen as well, which would allow us to remove the `VaspInputGenerator` class from `atomate2` entirely.

Reply:
Agreed this should be added to pymatgen.

Reply:
Moved to PMG